lib/spdy-transport/protocol/base/scheduler.js

```diff
@@ -121,7 +121,13 @@ Scheduler.prototype.tickSync = function tickSync () {
     debug('tick sync pending=%d', this.count, item.chunks)
     for (var j = 0; j < item.chunks.length; j++) {
       this.count--
-      res = this.push(item.chunks[j])
+      // TODO: handle stream backoff properly
```

Comment:
What does properly mean in this case?

Reply:
Right now we could be iterating over n+1 chunks. If the nth chunk causes this.push to return false, we're still going to try and push another chunk. This will cause an error and violates back pressure design in the streams api. As soon as false is returned we need to respect that and wait for _read to get called again.  I tried some other "quick fix" options, but it caused issues elsewhere. I need to spend more time getting up to speed on the transport to do a proper audit and refactor of the stream extensions.